lib/std/heap/general_purpose_allocator.zig

```diff
@@ -397,9 +399,11 @@ pub fn GeneralPurposeAllocator(comptime config: Config) type {
 
         /// Emits log messages for leaks and then returns whether there were any leaks.
         pub fn detectLeaks(self: *Self) bool {
+            self.ensureInitialized() catch unreachable;
```

Comment:
Uh, `catch unreachable` allocation failure is not ok.

Reply:
Would a panic be better or would a silent failure and return that there are no leaks be better?

Reply:
If the GPA is not initialized, surely we can immediately return that there were no leaks, since no memory was allocated?

Either way, yes, `catch unreachable` is not acceptable. `unreachable` means "control flow can never reach this bug; if it does, that is programmer error". Outside of specific cases like some uses of FBA, `error.OutOfMemory` is never an unreachable case.

Reply:
Essentially, solving this needs higher level design work for the GPA to meet the new constraint of page size being runtime known.

Reply:
What would that entail?

Reply:
Ok, I have it `return false` on failure now. Looks good?

Reply:
@ifreund Pinging you to see if the change is better since it's been some time since I've heard back.

Reply:
Answering the question of how this API should look would require me to do the higher level design work I mentioned, which I don't have time for.

Reply:
Well then the return false when the initialization fails in the leak detection should be good enough for this PR, could always update this once you or someone else has the time.